README.md

```diff
@@ -1,136 +1,181 @@
 </dependency>
-````
+```
+
+Add a feature file containing your scenarios in `src/test/resources/features`:
+
+```gherkin
+Feature: Hello World
+
+  Scenario: Hello World
+    Given the user name is 'World'
+    When the user clicks the hello button
+    Then the output is 'Hello World'
+```
+
+Implement the steps by creating a java class containing annotated methods:
+
+```java
+public class HelloWorldSteps {
+    private String name;
+    private String output;
+
+    @Given("^the user name is '([^']+)'$")
+    public void userNameIs(String name) {
+        this.name = name;
+    }
+
+    @When("^the user clicks the hello button$")
+    public void clickTheButton() {
+        this.output = "Hello " + name;
+    }
+
+    @Then("^the output is '([^']+)'")
+    public void outputIs(String expectedOutput) {
+        Assert.assertEquals("Output should be", expectedOutput, output);
+    }
+}
+```
+
+Implement a Junit runner for executing the features. The annotation parameters `featurePackages` and `stepDefinitionPackages` specify in which packages to search for features and step definitions respectively.
 
-After this is added you can start creating feature files with the standard [Gherkin](https://github.com/cucumber/cucumber/wiki/Gherkin) syntax.
-Step definitions can be added in the same way as with cucumber.
+```java
+@RunWith(Zuchini.class)
+@ZuchiniOptions(
+    featurePackages = {"features/helloworld"},
+    stepDefinitionPackages = {"org.zuchini.junit.helloworld"})
+public class HelloWorldTest {
+}
+```
 
-## Reporting
+## Reporting and test listeners
 
-To enable reporting add the zuchini-reporting dependency to your project.
-````        
+By default zuchini will report execution of scenarios without details about the individual steps. To enable reporting per step, set the `reportIndividualSteps` property in `ZuchiniOptions` to `true`.
+
+Zuchini allows to add additional `RunListener`'s to monitor test execution, using the `listeners` property of `ZuchiniOptions`. One such `RunListener` for generating test reports in json format is included in the `zuchini-reporting` artifact. 
+
+```xml
 <dependency>
     <groupId>org.zuchini</groupId>
     <artifactId>zuchini-reporting</artifactId>
-    <version>VERSION</version>
+    <version>${version.zuchini}</version>
     <scope>test</scope>
 </dependency>
-````
+```
 
-Additionally you have to configure your surefire plugin to write the report to a given file:
-````
+```java
+@RunWith(Zuchini.class)
+@ZuchiniOptions(
+    featurePackages = {"features/helloworld"},
+    stepDefinitionPackages = {"org.zuchini.junit.helloworld"},
+    listeners = {JsonReporter.class})
+public class HelloWorldTest {
+}
+```
+
+The report will be written to `zuchini-report.json` in a [format described by json schema](zuchini-reporter/schema.json).
+
+**Note**: The report format differs from the format used by cucumber-jvm and therefore cannot be interpreted by the usual report generation plugins.
```

Comment:
Is there a way to produce compatible reports (the default maybe)? I think that would be the place to highlight that.

Reply:
Not at the moment, I created issue #25 to track this.